test/validation/api/pktio/pktio.c

```diff
@@ -1195,10 +1195,8 @@ void pktio_test_index(void)
 
 	ndx = odp_pktio_index(pktio);
 	CU_ASSERT(ndx >= 0);
-	CU_ASSERT(odp_pktio_index(pktio_inval) < 0);
 
 	CU_ASSERT(odp_pktio_close(pktio) == 0);
-	CU_ASSERT(odp_pktio_index(pktio) < 0);
```

Comment:
Generally I would agree. However we explicitly have the invalid usecase in API:
```
@retval <0     On failure (e.g., handle not valid)
```

Reply:
We can have `_odp_pktio_index()` which does not perform such checks and `odp_pktio_index()` which does `ODP_PKTIO_INVALID` check.

Reply:
API does not (and should not) guarantee that invalid or stale handles are checked on fast path functions, such as this conversion to index. Especially, any API function is not specified to check against stale handles. Function may return -1, if it finds an error but we don't guarantee which errors are checked. For example, there may be check against invalid handle, but there may not be either. Invalid values are mostly needed for API function to be able return that e.g. create/open/alloc failed. E.g. here, application should check that pktio open succeeded, if it did not succeed it should not continue and pass that invalid handle to other pktio calls. That way every API call does not need to check the handle, only application needs to check it once after open/create/alloc.

This close + index test above it basically the same thing as doing:
odp_packet_free(pkt);
data = odp_packet_data(pkt);
// now we can assume that data is NULL, since pkt packet was freed and this function always searches through valid handles and notices that this handle value is not valid anymore.

Reply:
To me "(e.g. handle not valid)" does not yet guarantee the check. This would 
<0    When pktio handle is invalid, or some other error happened.

In general, we should remove all such "e.g. handle not valid" comments.

Reply:
@psavol Totally agree about close + index.